### DISCUSSION TITLE
Download of Images using the Storage class seems to have a problem

### DISCUSSION THREAD

**Discussion by chegejohn159** [laravel/framework#50780] [Q&A] [answered]
Hello team, I am trying to do downloads in laravell 11, pdfs and .txt work perfectly, but once I try with an Image (png, jpg, jpeg, jfif), The image downloads but trying to open it says file format not allowed. Could I be missing something?

**Comment by joy2362** [comment]
hello @chegejohn159  check your stroage folder if the image file save with correct extension or not if it is then provide some more info like code, file format etc ?

**Comment by Khey2** [comment]
Sorry for my bad English. I have the same problem, but I have 2 branches: in one it works, but in the other it does not. I have tried all day to find a solution, but I haven't.

I'm displaying the images in my application using a route.



<img width="628" alt="evidencia1" src="https://github.com/user-attachments/assets/36b7509f-ddd8-4fd7-85fc-9c431b7e6299">

and this is de de controller function 

<img width="511" alt="image" src="https://github.com/user-attachments/assets/ab9a6190-4612-4277-b37d-2fb90867d942">



Well, the strange thing here is the following: in my branch 1 (dev), this method works perfectly.

<img width="850" alt="image" src="https://github.com/user-attachments/assets/d74eae79-313d-4dd6-92a5-430286db93d2">


However, when I switch to the main branch, the image stops working.

<img width="491" alt="image" src="https://github.com/user-attachments/assets/e0a3862e-6ef8-4df6-bc83-dfee4bb75d04">

<img width="828" alt="image" src="https://github.com/user-attachments/assets/e097be2a-2ae0-442a-b074-8cb20d823879">


As you can see, it’s the same method. I have no idea why this is happening. I’ve been at it all day. As an additional note, in the branch where the image doesn’t show up, when I download it, it appears like this when opened from the file explorer.

<img width="724" alt="image" src="https://github.com/user-attachments/assets/18d83035-3d21-4fa6-8d4d-9672be328e08">


But, if I download this image from the other branch...

<img width="748" alt="image" src="https://github.com/user-attachments/assets/c8137c4a-d8d0-4fac-8519-c7515112245a">


i have tried all, i dont have any idea whats going on

**Reply by Khey2** [reply]
Extra, this problem ONLY HAPENDS WITH IMAGES, pdfs work perfectly

**Comment by chegejohn159** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/b367caf3-6a4d-48e3-b808-6d8ab1ef0fb0)


I had to do this for it to work
seems there were additional headers by default
here is the code
https://notepad.pw/download-laravel-11-img

On Fri, Nov 8, 2024 at 1:13 AM Khey2 ***@***.***> wrote:

> Extra, this problem ONLY HAPENDS WITH IMAGES, pdfs work perfectly
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/laravel/framework/discussions/50780#discussioncomment-11182846>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGE3IIRKVUVKFGUMEWJHH53Z7PQXLAVCNFSM6AAAAABRMGM4BKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJYGI4DINQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*John Chege Kamau*
Software Developer
+254 (0) 701 020 901
